### PR TITLE
Oop refactor

### PR DESCRIPTION
**Pull request by Beardless-sheik**
In this pull request, we: 
- [Added Display module to enable modularity of the Display functionality of the functions in the App class to follow the single responsibility principle 
- Fulfill Single-responsibility Principle by moving can_use_services? method to a new class - Services
- Removed unused functionality by following KISS programming

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -163,8 +165,11 @@ def list_rentals_by_person_id
...
 
   puts 'Rentals:'
-  @rentals.each do |rental|
-    puts "Date: #{rental.date}, Book '#{rental.book.title}' by #{rental.book.author}" if rental.person.id == id
+  Display.list(@rentals) do |rental|
+    puts "Date: #{rental.date}, Book   '#{rental.book.title}' by #{rental.book.author}" if rental.person.id == id
   end
+  # @rentals.each do |rental|
+  #   puts "Date: #{rental.date}, Book '#{rental.book.title}' by #{rental.book.author}" if rental.person.id == id
+  # end
```
Comment (Meltrust) [inline app.rb]: - [ ] Please, kindly delete this line of old commented code from the repo in order to improve its cleanliness 💪

File: app.rb [hunk 2]
```diff
@@ -4,8 +4,10 @@
 require_relative 'rental'
 require_relative 'book'
 require_relative 'classroom'
+require_relative 'display'
 
 class App
```
Comment (Meltrust) [inline app.rb]: - [ ] These methods are not following the [single responsibility principle](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for your APP class:

```RUBY
def create_person
    print 'Do you want to create a student (1) or teacher (2) [Input a number]: '
    option = gets.chomp

    case option
    when '1'
      create_student
    when '2'
      create_teacher
    else
      puts 'Invalid input. Kindly type 1 or 2'
    end
  end

  def create_student
    print 'Age: '
    age = gets.chomp.to_i

    print 'Name: '
    name = gets.chomp

    print 'Has parent permission? [Y/N]: '
    parent_permission = gets.chomp.downcase
    parent_permission = parent_permission == 'y'
    student = Student.new(@class, age, parent_permission, name)
    @people << student

    puts 'Student created successfully'
    sleep 0.75
  end

  def create_teacher
    print 'Age: '
    age = gets.chomp.to_i

    print 'Name: '
    name = gets.chomp

    print 'Specialization: '
    specialization = gets.chomp

    teacher = Teacher.new(specialization, age, name)
    @people << teacher

    puts 'Teacher created successfully'
    sleep 0.75
  end

  def create_book
    print 'Title: '
    title = gets.chomp

    print 'Author: '
    author = gets.chomp

    book = Book.new(title, author)
    @books << book

    puts 'Book added successfully'
    sleep 0.75
  end
end

def create_rental
  puts 'Select a book from the following list by number'
  @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }

  book_id = gets.chomp.to_i

  puts 'Select a person from the following list by number (not id)'
  @people.each_with_index do |person, index|
    puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
  end

  person_id = gets.chomp.to_i

  print 'Date: '
  date = gets.chomp

  rental = Rental.new(date, @people[person_id], @books[book_id])
  @rentals << rental

  puts 'Rental created successfully'
  sleep 0.75
end
```

Please, refer to the lesson and correct this issue so your APP class serves a single purpose 💪

Comment (Meltrust) [inline app.rb]: - [ ] Please, from these methods, kindly move these 2 indicated statements to your `create_person` method:

```RUBY
def create_student
    print 'Age: '
    age = gets.chomp.to_i

    print 'Name: '
    name = gets.chomp

    print 'Has parent permission? [Y/N]: '
    parent_permission = gets.chomp.downcase
    parent_permission = parent_permission == 'y'
    student = Student.new(@class, age, parent_permission, name) # THIS ONE (1)
    @people << student

    puts 'Student created successfully'
    sleep 0.75
  end

  def create_teacher
    print 'Age: '
    age = gets.chomp.to_i

    print 'Name: '
    name = gets.chomp

    print 'Specialization: '
    specialization = gets.chomp

    teacher = Teacher.new(specialization, age, name) # ...AND THIS ONE! (2)
    @people << teacher

    puts 'Teacher created successfully'
    sleep 0.75
  end
```

That way, your create_person method will create all of them 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @Beardless-sheik,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To highlight:

- Nice first implementation✔️ 

You can do it

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 2 above)
Comment (guerreiropedr0) [reply app.rb]: Hi, @Meltrust . Thank you for your review. I'm not quite sure how to solve it. Can you elaborate on that? I feel like they have only one responsibility. Can you please enlighten me?

Comment (guerreiropedr0) [reply app.rb]: We will gladly change it into one method only, but isn't it better to break into two since they are different? Can you please explain?

Comment (Meltrust) [reply app.rb]: Sure @guerreiropedr0!  Each of these methods has a single responsibility, but I'm talking about the class.  These methods don't belong there according to the Single responsibility rule (in the class).

Comment (Meltrust) [reply app.rb]: @guerreiropedr0 sorry, I was not clear enough on my snippets.  Only the line that creates a new teacher and the line that creates a new student, should be removed from those methods and included on `create_person`.

This one:

```ruby
student = Student.new(@class, age, parent_permission, name)
```

... and this one:

```ruby
teacher = Teacher.new(specialization, age, name)
```
I hope it's clearer now and makes sense to you.

Comment (Meltrust) [reply app.rb]: image

I'm talking about those lines, not the whole methods.

Comment (guerreiropedr0) [reply app.rb]: Oh, OK, I see it now. It's OK, you don't have to apologize. I think it's my fault that I didn't understand it right away. So thank you so much for explaining it so well.

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @Beardless-sheik and @guerreiropedr0,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**